workers/sentry/src/index.ts

```diff
@@ -0,0 +1,134 @@
+
+    this.validate(hawkEvent);
+
+    await this.addTask(WorkerNames.DEFAULT, hawkEvent as DefaultEventWorkerTask);
+  }
+
+  /**
+    * Parse Sentry envelope into structured format
+    *
+    * @param data - raw Sentry envelope data
+    * @returns Parsed SentryEnvelope object
+    */
+  private parseSentryEnvelope(data: string): SentryEnvelope {
+    const lines = data.split('\n').filter(line => line.trim() !== '');
+    const envelope: SentryEnvelope = { Header: {}, Items: [] };
+
+    // Parse envelope header
+    const headerLine = lines.shift();
+    if (!headerLine) {
+      throw new Error('Failed to read envelope header');
+    }
+    envelope.Header = JSON.parse(headerLine);
+
+    // Parse each item
+    while (lines.length > 0) {
+      // Item header
+      const itemHeaderLine = lines.shift();
+      if (!itemHeaderLine) {
+        throw new Error('Failed to read item header');
+      }
+      const itemHeader = JSON.parse(itemHeaderLine);
+
+      // Item payload
+      const itemPayloadLine = lines.shift();
+      if (!itemPayloadLine) {
+        throw new Error('Failed to read item payload');
+      }
+      const itemPayload = JSON.parse(itemPayloadLine);
+
+      envelope.Items.push({ Header: itemHeader, Payload: itemPayload });
+    }
+
+    return envelope;
+  }
+
+  /**
+   * Transform a Sentry event into a Hawk-compatible structure
+   *
+   * @param envelopeHeader - Sentry envelope header
+   * @param eventItem - Sentry event item
+   * @returns Hawk-compatible event structure
+   */
+  private transformToHawkFormat(
+    envelopeHeader: Record<string, any>,
+    eventItem: SentryItem,
+    projectId: string
+  ): DefaultEventWorkerTask {
+    const { sent_at, trace } = envelopeHeader;
+    const { Payload } = eventItem;
+
+    // delete public_key from trace
+    delete trace.public_key;
+
+    // convert sent_at from ISO 8601 to Unix timestamp
+    const sent_at_unix = Math.floor(new Date(sent_at).getTime() / 1000);
+
+    const backtrace = Payload.exception?.values?.[0]?.stacktrace?.frames?.map((frame: any) => ({
+      file: frame.filename,
+      line: frame.lineno,
+      function: frame.function,
+      sourceCode: frame.pre_context?.concat(frame.context_line, frame.post_context || [])
+        .map((line: string, index: number) => ({
+          line: frame.lineno + index - frame.pre_context.length,
+          content: line
+        })) || [],
+    })) || [];
+
+    return {
+      projectId: projectId, // Public key is used as hawk project ID
+      catcherType: 'errors/sentry',
+      payload: {
+        title: `${Payload.exception?.values?.[0]?.type || 'Unknown'}: ${Payload.exception?.values?.[0]?.value || ''}`,
+        type: Payload.level || 'error',
+        timestamp: sent_at_unix,
+        backtrace,
+        release: trace?.release || null,
+        context: {
+          sentAt: sent_at_unix,
+          trace: trace || {},
+          environment: trace?.environment || 'unknown',
+        },
```

Comment:
do sentry sends additional data like our "context" field?

Reply:
yep! Context is sent via trace param.
```
context: {
          sentAt: sent_at_unix,
          trace: trace || {},
          environment: trace?.environment || 'unknown',
        },
```

trace sample is on the screenshot
<img width="863" alt="image" src="https://github.com/user-attachments/assets/e581dc53-ff90-4c9b-bb97-101c46da3824" />